storage/imx6ull-flash/imx6ull-flash-test.c

```diff
@@ -112,6 +114,164 @@ void print_block(const uint8_t *data, unsigned int len)
 }
 
 
+/* Flips n bits in a raw block (assume n <= end - start) */
+int flip_bits(unsigned int n, unsigned int nblock, unsigned int start, unsigned int end)
+{
+	unsigned int i, j, k, offs, paddr = nblock * BLOCK_PAGES_CNT;
+	flashdrv_dma_t *dma;
+	flashdrv_meta_t *aux;
+	uint8_t **data;
+	int err = EOK;
+
+	if ((dma = flashdrv_dmanew()) == MAP_FAILED)
+		return -ENOMEM;
+
+	if ((data = malloc(BLOCK_PAGES_CNT * sizeof(uint8_t *))) == NULL) {
+		flashdrv_dmadestroy(dma);
+		return -ENOMEM;
+	}
+
+	do {
+		/* Read raw block data into memory and flip bits */
+		for (i = 0; i < BLOCK_PAGES_CNT; i++) {
+			if ((data[i] = mmap(NULL, (FLASHDRV_PAGESZ + _PAGE_SIZE - 1) & ~(_PAGE_SIZE - 1), PROT_READ | PROT_WRITE, MAP_UNCACHED, OID_CONTIGUOUS, 0)) == MAP_FAILED) {
+				err = -ENOMEM;
+				printf("mmap() failed: %d\n", err);
+				break;
+			}
+
+			if ((err = flashdrv_readraw(dma, paddr + i, data[i], FLASHDRV_PAGESZ)) < 0) {
+				printf("readraw() failed: %d\n", err);
+				break;
+			}
+
+			for (offs = i * FLASHDRV_PAGESZ; start < min(end, offs + FLASHDRV_PAGESZ); start += (end - start) / n--)
```

Comment:
On the contrary, the idea is to have a single bit flip per byte evenly spaced out across the [`start`; `end`) range. I changed the code so the intention is more clear.

Reply:
Ok, now I see it, clever. Probably computing `step = (end - start) / n` once and reusing it would be much more readable (no additional comment would be necessary then), but let's leave it as-is ;).